loris/img_info.py

```diff
@@ -145,133 +144,130 @@ def _from_jp2(self, fp):
         scaleFactors = []
 
         #TODO use context manager so it's automatically closed
-        jp2 = open(fp, 'rb')
-
-        #check that this is a jp2 file
-        initial_bytes = jp2.read(24)
-        if (not initial_bytes[:12] == '\x00\x00\x00\x0cjP  \r\n\x87\n') or \
-            (not initial_bytes[16:] == 'ftypjp2 '):
-            jp2.close()
```

Comment:
FFS, GitHub diffs. All I’ve done is indented this entire block and dropped this `jp2.close()` line – it’s unnecessary with the context manager.